app/actions/CisUserDataFilterAction.scala

```diff
@@ -36,8 +36,8 @@ case class CisUserDataFilterAction(taxYear: Int,
   override protected[actions] def filter[A](input: UserSessionDataRequest[A]): Future[Option[Result]] = {
     val result = input.cisUserData match {
       case data if data.isPriorSubmission && redirectIfPrior => Some(Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear)))
-      case data if data.hasPeriodData || !needsPeriodData => None
-      case data if !data.hasPeriodData => Some(Redirect(DeductionPeriodController.show(taxYear, employerRef)))
+      case data if !data.hasPeriodData && needsPeriodData => Some(Redirect(DeductionPeriodController.show(taxYear, employerRef)))
```

Comment:
This seems correct, however I need to think more about it. The usages of this class need to be reviewed. I think I added comments in the past that this needs refactoring.
I would for now leave it as it was and we can come back to this to ensure we don't brake something.